sycl/doc/extensions/DeviceGlobal/SPV_INTEL_global_variable_decorations.asciidoc

```diff
@@ -0,0 +1,231 @@
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_global_variable_decorations"
+----
+
+== New Capabilities
+This extension introduces a new capability:
+
+----
+GlobalVariableDecorationsINTEL
+----
+
+== Token Number Assignments
+
+--
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|GlobalVariableDecorationsINTEL | 6146
+|HostAccessINTEL | 6147
+|InitModeINTEL | 6148
+|ImplementInCSRINTEL | 6149
+|====
+--
+
+== Modifications to the SPIR-V Specification, Version 1.5
+
+=== Decoration
+
+Modify Section 3.20, Decoration, adding these rows to the Decoration table:
+
+--
+[options="header"]
+|====
+2+^| Decoration 2+^| Extra Operands | Enabling Capabilities
+
+// --- ROW BREAK ---
+| 6147
+a|
+*HostAccessINTEL* +
+Only valid on global (module scope) *OpVariable*.
+
+The client API's execution environment may provide a way to access a global
+variable's value from the host system.  If it does, this decoration provides
+two pieces of information.  _Access_ is an assertion by the producer about the
+types of these accesses, which may allow the consumer to perform certain
+optimizations, especially for FPGA devices.  _Name_ is a name which the client
+API's execution environment may use to identify this variable.
+
+Legal values of _Access_:
+
+* 0 [Read] - The execution environment may read the variable from the host but
+  will never write it.  On an FPGA device, only a read memory port is exposed.
+* 1 [Write] - The execution environment may write the variable from the host
+  but will never read it.  On an FPGA device, only a write memory port is
+  exposed.
+* 2 [Read/Write] - The execution environment may read or write the variable
+  from the host.  On an FPGA device, a read/write memory port is exposed.
+* 3 [None] - The execution environment may neither read nor write the variable
+  from the host.  On an FPGA device, no memory port is exposed.
+
+If a global *OpVariable* is not decorated with *HostAccessINTEL*, the default
+behavior is [Read/Write].
+| Literal Number +
+_Access_
+| Literal String +
+_Name_
+| *GlobalVariableDecorationsINTEL*
+
+// --- ROW BREAK ---
+| 6148
+a|
+*InitModeINTEL* +
+Only valid on global (module scope) *OpVariable* which has an _Initializer_
+operand.
+
+This decoration only has an effect when the consumer is an FPGA or similar
+device.  The _Trigger_ value tells how the global variable should be
+initialized.
+
+Legal values of _Trigger_:
+
+* 0 [init on device reprogram] - Initialization is performed by reprogramming
+  the device.  This may require more frequent reprogramming but may reduce
+  area.
+* 1 [init on device reset] - Initialization is performed by sending a reset
+  signal to the device.  This may increase area but may reduce reprogramming
+  frequency.
+
+If a global *OpVariable* with an _Initializer_ operand is not decorated with
+*InitModeINTEL*, the method by which its value is initialized is implementation
```

Comment:
"... the method by which **the device** is initialized is implementation defined."

I like this better. It uses the same language as the description of this decoration, above.